### DISCUSSION TITLE
html-text library

### DISCUSSION THREAD

**Discussion by mizaki** [comictagger/comictagger#394] [Ideas]
I gave https://pypi.org/project/html-text/ a try with Batman issue id 529618:
```
<p><em>Echoes from Batman's past ripple out into Gotham City's future…</em></p><h4>List of covers and their creators:</h4><table data-max-width=\"true\"><thead><tr><th scope=\"col\">Cover</th><th scope=\"col\">Name</th><th scope=\"col\">Creator(s)</th><th scope=\"col\">Sidebar Location</th></tr></thead><tbody><tr><td>Reg</td><td>Regular Cover</td><td>Greg Capullo, Danny Miki &amp; FCO Plascencia</td><td>1</td></tr><tr><td>Var</td><td>The New 52 Homage Variant Cover</td><td>Rafael Albuquerque</td><td>2</td></tr></tbody></table><p><strong>Note</strong>: Riddler, Wesker, Professor Pyg, Scarecrow, Two-Face and Clayface only appear on the variant cover.</p>
```
Result with default options:
```
Echoes from Batman's past ripple out into Gotham City's future…

List of covers and their creators:

Cover Name Creator(s) Sidebar Location
Reg Regular Cover Greg Capullo, Danny Miki & FCO Plascencia 1
Var The New 52 Homage Variant Cover Rafael Albuquerque 2

Note: Riddler, Wesker, Professor Pyg, Scarecrow, Two-Face and Clayface only appear on the variant cover.
```
There is the option to remove tags from single or double returns in a set list: `html_text.NEWLINE_TAGS` and `html_text.DOUBLE_NEWLINE_TAGS`

With the removal of "p" and "h4" the result is:
```
Echoes from Batman's past ripple out into Gotham City's future… List of covers and their creators:
Cover Name Creator(s) Sidebar Location
Reg Regular Cover Greg Capullo, Danny Miki & FCO Plascencia 1
Var The New 52 Homage Variant Cover Rafael Albuquerque 2
Note: Riddler, Wesker, Professor Pyg, Scarecrow, Two-Face and Clayface only appear on the variant cover.
```
I think this library looks prefect? The only thing is it will always remove the tables but I'm unsure why you would ever want to keep them and remove the rest of the HTML? Maybe a better option would be to offer leaving the description untouched for those who want the HTML?

**Comment by lordwelch** [comment]
I would rather make these changes than switch to this library. html-text makes no attempt to make a table look like a table once it is converted. Where as with these changes it now uses proper markdown.
[diff.txt](https://github.com/comictagger/comictagger/files/10038386/diff.txt)
